### ISSUE TITLE
GeoDataFrame does not maintain input source's schema

### ISSUE THREAD

**Issue by WeatherGod** [geopandas/geopandas#177]
I am trying to use geopandas to make some simple shapefile modification scripts (kinda like the ogr2ogr type stuff, but tailored for our production pipeline. I have found that loading a shapefile does not retain the shapefile schema.

```
>>> import geopandas as gpd
>>> a = gpd.read_file('testinput.shp')
>>> a.to_file('testoutput.shp')
```
```
$ dbfdump -h testinput.dbf | head -n 9
Field 0: Type=N/Double, Title=`startHrGMT', Width=7, Decimals=3
Field 1: Type=D/String, Title=`startDate', Width=8, Decimals=0
Field 2: Type=N/Double, Title=`endHrGMT', Width=7, Decimals=3
Field 3: Type=D/String, Title=`endDate', Width=8, Decimals=0
Field 4: Type=C/String, Title=`descript', Width=30, Decimals=0
Field 5: Type=C/String, Title=`dataRights', Width=92, Decimals=0
Field 6: Type=C/String, Title=`units', Width=10, Decimals=0
Field 7: Type=N/Integer, Title=`Poly_Id', Width=6, Decimals=0
Field 8: Type=N/Integer, Title=`HailProb', Width=10, Decimals=0
$ dbfdump -h testoutput.dbf | head -n 9
Field 0: Type=N/Integer, Title=`HailProb', Width=10, Decimals=0
Field 1: Type=N/Integer, Title=`Poly_Id', Width=10, Decimals=0
Field 2: Type=C/String, Title=`dataRights', Width=91, Decimals=0
Field 3: Type=C/String, Title=`descript', Width=80, Decimals=0
Field 4: Type=C/String, Title=`endDate', Width=80, Decimals=0
Field 5: Type=N/Double, Title=`endHrGMT', Width=24, Decimals=15
Field 6: Type=C/String, Title=`startDate', Width=80, Decimals=0
Field 7: Type=N/Double, Title=`startHrGMT', Width=24, Decimals=15
Field 8: Type=C/String, Title=`units', Width=80, Decimals=0
```
I don't mind that the fields are not in the same order, but the data types are wrong (the D/String for endDate and startDate are now C/String), and the width and precision are all different now too. We have customers who load this data into preallocated memory structures, so this is a non-starter for us.

**Comment by fmaussion** [comment]
Is this issue likely to occur on other shapefiles too, or is it specific to WeatherGod's? I can't seem to reproduce the problem with mine but this would be a problem if it happens without warning on another file. Thanks!

**Comment by WeatherGod** [comment]
It always happens, but you might not always realize it. For example, if
your shapefile's schema perfectly matches the inferred schema from the
data, then the round trip will "maintain" the schema, but only by accident.
So, if your data is full of fixed width strings, and the schema is set up
to be that fixed width, then you won't notice any difference. Also, if you
never defined the width and/or precision of your integer or float columns,
then you won't notice a difference since it is merely falling back onto the
default width and precision.

**Comment by WeatherGod** [comment]
This issue is sufficiently addressed by #178

**Comment by adrivsh** [comment]
Hi

I do think it would be useful that geopandas caries the schema. It has a simplify method, which means many will want to open a shape file, simplify it, and export it. You need to carry the schema for that.

**Comment by ligz08** [comment]
+1 for carrying schema with GeoDataFrame.

I was using geopandas to perform some analysis on a shapefile, adding several new columns, and save as another shapefile with `to_file()`.
I had to manually fetch schema via `fiona.open` and add schema items for the new columns I created, otherwise GIS software like ArcGIS wouldn't recognize the exported `.dbf` attribute table.

In my particular case, geopandas exported attributes that used to be of `float:19.10` type as `float:24.15` type.
